### PR TITLE
Separate create entity doc into entity template and world commands.

### PR DESCRIPTION
**Pull request by martinzlocha**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
This PR separates the existing create entity doc into an entity template doc which can be referenced from both MB and ECS world commands docs. Most of the text has been copied over so I have created a PR directly. If this proves to be an issue I will go back and create a GDoc.

#### Tests
How did you test these changes prior to submitting this pull request?
What automated tests are included in this PR?
N/A

#### Documentation
How is this documented (for example: release note, upgrade guide, feature page, in-code documentation)?
N/A

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: docs/content/entity-templates.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+------
+
+## Creating entity templates
+
+To create an [entity](https://docs.improbable.io/reference/latest/shared/glossary#entity) in SpatialOS, you need to create a template which specifies which components should be added to the entity and allows you to specify [authority](authority.md) on a per-[component](https://docs.improbable.io/reference/latest/shared/glossary#component) basis.
+
+The entity template can be created using the `EntityBuilder` which is part of the `Improbable.Gdk.Core` assembly. The `EntityBuilder` class has the following public methods:
+
+The `EntityBuilder` class has the following public methods:
```
Comment (JonasImprobable) [inline docs/content/entity-templates.md]: Duplicate sentence.

File: docs/content/ecs/world-commands.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        {
+            var entity = CreatureTemplate.CreateCreatureEntityTemplate(
+            	new Coordinates(0, 0, 0));
+
+            var request = new WorldCommands.CreateEntity.CreateRequest
+            (
+                entity
+            ));
+
+            data.CreateEntitySender[i].RequestsToSend.Add(request);
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: We should get in the habit of doing set backs even with lists.

File: docs/content/ecs/world-commands.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+To receive a response use `WorldCommands.DeleteEntity.CommandResponses`. This contains a list of `WorldCommands.DeleteEntity.ReceivedResponse`.
+
+```csharp
+public class DeleteCreatureSystem : ComponentSystem
+{
+    public struct Data
+    {
+        public readonly int Length;
+        public ComponentDataArray<Bar> Bar;
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: `[ReadOnly]`

File: docs/content/ecs/world-commands.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+You can use entity queries to get information about entities in the world. 
+
+To send a request use `WorldCommands.EntityQuery.CommandSender`. This contains a list of `WorldCommands.EntityQuery.Request` structs. Add a struct to the list to send the command
+
+- For more information, see [entity queries](https://docs.improbable.io/reference/latest/shared/glossary#queries) in the SpatialOS documentation.
+- `TimeoutMillis` is optional.
+
+To receive a response use `WorldCommands.EntityQuery.CommandResponses`. This contains a list of `WorldCommands.EntityQuery.ReceivedResponse`.
+
+**Warning**: Entity queries only exist in the GDK in a prototype form, the methods for sending and receiving them exist, but you can not yet safely access the information in the responses. It is not recommended to use them.
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: You can access _some_ information in the response. Not sure if we want to be more nuanced

File: docs/content/entity-templates.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
+**Warning:** The [alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages) release is for evaluation purposes only, with limited documentation - see the guidance on [Recommended use](../../../README.md#recommended-use).
+
+------
+
+## Creating entity templates
+
+To create an [entity](https://docs.improbable.io/reference/latest/shared/glossary#entity) in SpatialOS, you need to create a template which specifies which components should be added to the entity and allows you to specify [authority](authority.md) on a per-[component](https://docs.improbable.io/reference/latest/shared/glossary#component) basis.
```
Comment (jamiebrynes7) [inline docs/content/entity-templates.md]: `in the SpatialOS GDK for Unity` vs `in SpatialOS`

File: docs/content/entity-templates.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+There is a method `Health.Component.CreateSchemaComponentData(int healthValue)` which will instantiate and populate a `ComponentData` object.
+
+Below is an example of how to define an entity template using the `EntityBuilder`. In this example we will be creating a template for a creature:
+
+```csharp
+public static class CreatureTemplate
+{
+    public static Entity CreateCreatureEntityTemplate(Coordinates coords)
+    {
+        const string entityType = "Creature";
```
Comment (jamiebrynes7) [inline docs/content/entity-templates.md]: Just inline this in the metadata

File: docs/content/entity-templates.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+            .SetReadAcl(WorkerUtils.AllWorkerAttributes)
+            .AddComponent(healthComponent, WorkerUtils.UnityGameLogic)
+            .Build();
+
+        return entity;
+    }
+}
+```
+
+We have created an entity template, now we should spawn this entity. This can be done either using the [MonoBehaviour](gameobject/world-commands.md) or [ECS](ecs/world-commands.md) workflow.
```
Comment (jamiebrynes7) [inline docs/content/entity-templates.md]: Reword first sentence to `You can use this entity template to spawn a Creature entity.` or similar

File: docs/content/ecs/world-commands.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+To send a request use `WorldCommands.ReserveEntityIds.CommandSender`. This contains a list of `WorldCommands.ReserveEntityIds.Request` structs. Add a struct to the list to send the command.
+
+- `TimeoutMillis` is optional.
+
+To receive a response use `WorldCommands.ReserveEntityIds.CommandResponses`. This contains a list of `WorldCommands.ReserveEntityIds.ReceivedResponse` structs.
+
+### 2. Create an entity
+
+You can use the  `CreateEntity` world command to request the creation of a new SpatialOS entity which you specified using an [entity template](../entity-templates.md).
```
[the remark this answers is not in the document]
Comment (martinzlocha) [reply docs/content/ecs/world-commands.md]: I used `specified` because its a doc which they completed earlier. But not sure if this makes sense in English.

**Comment by martinzlocha** [comment]
@jamiebrynes7 @JonasImprobable Made changes based on your comments :) Please have a look again.

File: docs/content/entity-templates.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
+**Warning:** The [alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages) release is for evaluation purposes only, with limited documentation - see the guidance on [Recommended use](../../../README.md#recommended-use).
+
+------
+
+## Creating entity templates
+
+To create an [entity](https://docs.improbable.io/reference/latest/shared/glossary#entity) in the SpatialOS GDK for Unity, you need to create a template which specifies which components should be added to the entity and allows you to specify [authority](authority.md) on a per-[component](https://docs.improbable.io/reference/latest/shared/glossary#component) basis.
```
Comment (jessicafalk) [inline docs/content/entity-templates.md]: authority link not working

File: docs/content/entity-templates.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+| `SetEntityAclComponentWriteAccess(string attribute)`         | Specify which worker type can have authority over the [`EntityAcl`](https://docs.improbable.io/reference/latest/shared/schema/standard-schema-library#entityacl-required) component of your entity. This is useful if, while the game is running, you want to change which worker type(s) can have authority over the entity's other components. |
+| `AddComponent(ComponentData data, string writeAccess)`       | Add a user-defined component to your entity and specify which worker type(s) can have authority over it. |
+| `Build()`                                                    | Create a finished entity template.                           |
+
+For each [schema component](https://docs.improbable.io/reference/13.2/shared/glossary#schema) you define, [the code generator](./code-generator.md) generates a helper method to create a `ComponentData` object for that component. For example, for the following schema:
```
Comment (jessicafalk) [inline docs/content/entity-templates.md]: code-gen link also not working

File: docs/content/ecs/world-commands.md [hunk 11]
```diff
@@ -0,0 +1,126 @@
+**Warning:** The [alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages) release is for evaluation purposes only, with limited documentation - see the guidance on [Recommended use](../../../README.md#recommended-use).
+
+-----
+
+
+## ECS: World commands
+
+World commands are special commands that are sent to the SpatialOS runtime to ask it to reserve entity ids, create or delete entities, or request information about entities. (See the SpatialOS documentation on [world commands](https://www.google.com/url?q=https://docs.improbable.io/reference/latest/shared/design/commands%23world-commands&sa=D&ust=1536752675413000) for more information.) 
+
```
Comment (ElleEss) [inline docs/content/ecs/world-commands.md]: This link is a bit weird.

File: docs/content/ecs/world-commands.md [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+        }
+    }
+}
+```
+
+This system iterates through every entity with a `Foo` component and sends a create entity request.
+
+### 3.  Delete an entity
+
+Deleting entities can be done similarly via the `DeleteEntity` world command. You need to know the SpatialOS entity ID of the entity you want to delete.
```
Comment (ElleEss) [inline docs/content/ecs/world-commands.md]: I know this is a copy and paste but as we are here - this switching to passive voice makes it unclear who is going this.
Suggest:
"Deleting entities can be done similarly via the `DeleteEntity` world command."
 -> amend to..
"You can delete entities via the `DeleteEntity` world command."

File: docs/content/ecs/world-commands.md [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+This system iterates through every entity with a `Bar` component and sends a delete entity request.
+
+### 4. Entity query
+
+You can use entity queries to get information about entities in the world. 
+
+To send a request use a `WorldCommands.EntityQuery.CommandSender` component. This contains a list of `WorldCommands.EntityQuery.Request` structs. Add a struct to the list to send the command
+
```
Comment (ElleEss) [inline docs/content/ecs/world-commands.md]: command.  <- full stop needed.

File: docs/content/entity-templates.md [hunk 14]
```diff
@@ -0,0 +1,58 @@
+**Warning:** The [alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages) release is for evaluation purposes only, with limited documentation - see the guidance on [Recommended use](../../../README.md#recommended-use).
+
+------
+
+## Creating entity templates
+
+To create an [entity](https://docs.improbable.io/reference/latest/shared/glossary#entity) in the SpatialOS GDK for Unity, you need to create a template which specifies which components should be added to the entity and allows you to specify [authority](authority.md) on a per-[component](https://docs.improbable.io/reference/latest/shared/glossary#component) basis.
+
+The entity template can be created using the `EntityBuilder` which is part of the `Improbable.Gdk.Core` assembly. The `EntityBuilder` class has the following public methods:
```
Comment (ElleEss) [inline docs/content/entity-templates.md]: "The entity template can be created using the `EntityBuilder`..."
->
"Create the entity template using the `EntityBuilder`..."

(The original is in passive voice so I reader has to infer that it's they who have to do something. Our text doesn't explicitly tell them.)

**Comment by martinzlocha** [comment]
@ElleEss @jessicafalk Made changes based on your feedback :)

File: docs/content/entity-templates.md [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+| `SetEntityAclComponentWriteAccess(string attribute)`         | Specify which worker type can have authority over the [`EntityAcl`](https://docs.improbable.io/reference/latest/shared/schema/standard-schema-library#entityacl-required) component of your entity. This is useful if, while the game is running, you want to change which worker type(s) can have authority over the entity's other components. |
+| `AddComponent(ComponentData data, string writeAccess)`       | Add a user-defined component to your entity and specify which worker type(s) can have authority over it. |
+| `Build()`                                                    | Create a finished entity template.                           |
+
+For each [schema component](https://docs.improbable.io/reference/13.2/shared/glossary#schema) you define, [the code generator](code-generator.md) generates a helper method to create a `ComponentData` object for that component. For example, for the following schema:
```
Comment (jessicafalk) [inline docs/content/entity-templates.md]: code generator link not working

File: docs/content/ecs/world-commands.md [hunk 16]
```diff
@@ -0,0 +1,126 @@
...
+## ECS: World commands
+
+World commands are special commands that are sent to the SpatialOS runtime to ask it to reserve entity ids, create or delete entities, or request information about entities. (See the SpatialOS documentation on [world commands](https://docs.improbable.io/reference/latest/shared/design/commands#world-commands) for more information.) 
+
+Each ECS entity that represents a SpatialOS entity has a set of components for sending world commands. For each world command, there is a component to send the command and receive the response. 
+
+### 1. Reserve an entity ID
+
+You can use the `ReserveEntityIds` world command before entity creation which makes it easier to create multiple entities in a group.
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: Would reword this to:

> You can use the `ReserveEntityIds` world command to reserve groups of entity IDs that you can use in entity creation.

Or something similar - it doesn't really make it easier to create entities in a group.

File: docs/content/ecs/world-commands.md [hunk 17]
```diff
@@ -0,0 +1,126 @@
...
+            var requests = data.CreateEntitySender[i].RequestsToSend;
+            var entity = CreatureTemplate.CreateCreatureEntityTemplate(
+                new Coordinates(0, 0, 0));
+
+            requests.Add(new WorldCommands.CreateEntity.CreateRequest
+            (
+                entity
+            ));
+
+            data.CreateEntitySender[i].requestsd;
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: `requestsd` 😛

File: docs/content/ecs/world-commands.md [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+        public readonly int Length;
+        [ReadOnly] public ComponentDataArray<Foo> Foo;
+        public ComponentDataArray<WorldCommands.CreateEntity.CommandSender> CreateEntitySender;
+    }
+
+    [Inject] Data data;
+
+    protected override void OnUpdate()
+    {
+        for(var i = 0; i < data.Length; i++)
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: Super nit: Whitespace between `for` and `(`

File: docs/content/ecs/world-commands.md [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+        {
+            var requests = data.DeleteEntitySender[i].RequestsToSend;
+            var entityId = data.SpatialEntityIds[i].EntityId;
+        	
+            requests.Add(new WorldCommands.DeleteEntity.CreateRequest
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: Drop the `new`

File: docs/content/ecs/world-commands.md [hunk 20]
```diff
@@ -0,0 +1,126 @@
...
+
+    protected override void OnUpdate()
+    {
+        for(var i = 0; i < data.Length; i++)
+        {
+            var requests = data.CreateEntitySender[i].RequestsToSend;
+            var entity = CreatureTemplate.CreateCreatureEntityTemplate(
+                new Coordinates(0, 0, 0));
+
+            requests.Add(new WorldCommands.CreateEntity.CreateRequest
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: Drop the `new`

File: docs/content/ecs/world-commands.md (hunk 18 above)
Comment (jamiebrynes7) [reply docs/content/ecs/world-commands.md]: Similar for the other code examples

File: docs/content/ecs/world-commands.md [hunk 21]
```diff
@@ -0,0 +1,126 @@
...
+                entityId
+            ));
+
+            data.DeleteEntitySender[i].RequestsToSend = requests;
+        }
+    }
+}
+```
+
+This system iterates through every entity with a `Bar` component and sends a delete entity request.
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: > ...every entity with a `Bar` **and a `SpatialEntityId`** component...

File: docs/content/ecs/world-commands.md [hunk 22]
```diff
@@ -0,0 +1,126 @@
...
+You can use entity queries to get information about entities in the world. 
+
+To send a request use a `WorldCommands.EntityQuery.CommandSender` component. This contains a list of `WorldCommands.EntityQuery.Request` structs. Add a struct to the list to send the command.
+
+- For more information, see [entity queries](https://docs.improbable.io/reference/latest/shared/glossary#queries) in the SpatialOS documentation.
+- `TimeoutMillis` is optional.
+
+To receive a response use `WorldCommands.EntityQuery.CommandResponses`. This contains a list of `WorldCommands.EntityQuery.ReceivedResponse`.
+
+**Warning**: Entity queries only exist in the GDK in a prototype form, the methods for sending and receiving them exist, but you can only access some information  safely in the responses. It is not recommended to use them.
```
Comment (jamiebrynes7) [inline docs/content/ecs/world-commands.md]: It is not recommended to use them **at this time**. 

Makes it clear that we are intending to fix them

File: docs/content/entity-templates.md [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+            .SetReadAcl(WorkerUtils.AllWorkerAttributes)
+            .AddComponent(healthComponent, WorkerUtils.UnityGameLogic)
+            .Build();
+
+        return entity;
+    }
+}
+```
+
+You can use this entity template to spawn a Creature entity. This can be done either using the [MonoBehaviour](gameobject/world-commands.md) or [ECS](ecs/world-commands.md) workflow.
```
Comment (jamiebrynes7) [inline docs/content/entity-templates.md]: Drop the `either`.

"workflow for World Commands." ..?
